src/Kestrel/ServerRequest.cs

```diff
@@ -231,7 +233,53 @@ bool IHttpUpgradeFeature.IsUpgradableRequest
                 }
             }
             _frame.ProduceStart();
-            return _frame.DuplexStream;
+            return Task.FromResult<Stream>(_frame.DuplexStream);
+        }
+
+        IPAddress IHttpConnectionFeature.RemoteIpAddress
+        {
+            get
+            {
+                IPAddress ip;
+                var result = IPAddress.TryParse(_frame.RemoteIpAddress, out ip);
+                return result ? ip : IPAddress.Any;
+            }
+            set
+            {
+                throw new NotSupportedException();
```

Comment:
You cannot set them since these are bound when a connection is made. It is up to Libuv to select them when the connection is made.